AddonBicepTemplate/AzureMonitorAlertsProfile.bicep

```diff
@@ -268,8 +268,8 @@ resource communityALerts 'Microsoft.AlertsManagement/prometheusRuleGroups@2023-0
         ]
       }
       {
-        alert: 'KubePodNotReady'
-        expression: 'sum by (namespace, pod, cluster) (  max by(namespace, pod, cluster) (    kube_pod_status_phase{job="kube-state-metrics", phase=~"Pending|Unknown"}  ) * on(namespace, pod, cluster) group_left(owner_kind) topk by(namespace, pod, cluster) (    1, max by(namespace, pod, owner_kind, cluster) (kube_pod_owner{owner_kind!="Job"})  )) > 0'
+        alert: 'KubePodNotReadyPerController'
```

Comment:
Soham, can you pls change the name to 'KubePodNotReadyByController' ? Please change the description of the alert to say its grouped by controller name. Have you also checked 2 controlers with same name in 2 different namespaces and see if the alert does indeed generate seperate alerts for those 2 controllers ?
Also what happens to controller less raw pods ? It will alert with an empty controller name ?